var/spack/repos/builtin/packages/py-pycuda/package.py

```diff
@@ -24,6 +24,26 @@ class PyPycuda(PythonPackage):
 
     depends_on("cxx", type="build")  # generated
 
+    # TODO: replace this with an explicit list of components of Boost,
+    # for instance depends_on('boost +filesystem')
+    # See https://github.com/spack/spack/pull/22303 for reference
+    depends_on(Boost.with_default_variants)
+    depends_on("python@3.6:3", type=("build", "run"), when="@2020.1:")
+    depends_on("python@3.8:3", type=("build", "run"), when="@2024.1.2:")
+    depends_on("boost+python")
+    depends_on("cuda", type=("build", "run"))
+    depends_on("cuda@:8.0.61", when="@2016.1.2")
+    depends_on("py-appdirs@1.4.0:", type=("build", "run"))
```

Comment:
Unless you know otherwise, `py-appdirs` no longer appears to apply for the package. IDK which version it was dropped in -- ideally you could check -- but assuming it was dropped in version `2023.y.z`, you could constrain this as:

```suggestion
    depends_on("py-appdirs@1.4.0:", type=("build", "run"), when="@:2023")
```